core/src/main/scala/org/ensime/core/Project.scala

```diff
@@ -12,10 +12,13 @@ import org.ensime.indexer._
 
 import scala.collection.immutable.ListSet
 import scala.concurrent.duration._
+import scala.util.Properties._
 import scala.util._
 import org.ensime.util.file._
 import org.ensime.util.FileUtils
 
+case object CloseOnIndexUpdate
```

Comment:
or there might be an idiomatic akka way to tell an actor to shutdown and take the system with it.